plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1204 @@
+	 *		// Filter the entire items array so only the items that start
+	 *		// with the query remain.
+	 *		var suggestions = itemsArray.filter( function( item ) {
+	 *			return item.name.indexOf( query ) === 0;
+	 *		} );
+	 *
+	 *		// Note - the callback function can also be executed asynchronously
+	 *		// so dataCallback can do an XHR requests or use any other asynchronous API.
+	 *		callback( suggestions );
+	 *	}
+	 *
+	 *	// Finally, instantiate the autocomplete class.
+	 *	new CKEDITOR.plugins.autocomplete( editor, textTestCallback, dataCallback );
+	 * ```
+	 *
+	 * ## Changing the behavior of the autocomplete class by subclassing it
+	 *
+	 * This plugin will expose a `CKEDITOR.plugins.customAutocomplete` class which uses
+	 * a custom view that positions the panel relative to the {@link CKEDITOR.editor#container}.
+	 *
+	 * ```javascript
+	 *	CKEDITOR.plugins.add( 'customautocomplete', {
+	 *		requires: 'autocomplete',
+	 *
+	 *		onLoad: function() {
+	 *			var View = CKEDITOR.plugins.autocomplete.view,
+	 *				Autocomplete = CKEDITOR.plugins.autocomplete;
+	 *
+	 *			function CustomView( editor ) {
+	 *				// Call the parent class constructor.
+	 *				View.call( this, editor );
+	 *			}
+	 *			// Inherit the view methods.
+	 *			CustomView.prototype = CKEDITOR.tools.prototypedCopy( View.prototype );
+	 *
+	 *			// Change the positioning of the panel, so it is stretched
+	 *			// to 100% of the editor container width and is positioned
+	 *			// relative to the editor container.
+	 *			CustomView.prototype.updatePosition = function() {
+	 *				var caretRect = this.getCaretRect(),
+	 *					container = this.editor.container;
+	 *
+	 *				this.setPosition( {
+	 *					// Position the panel relative to the editor container.
+	 *					left: container.$.offsetLeft,
+	 *					top: caretRect.top,
+	 *					bottom: caretRect.bottom
+	 *				} );
+	 *				// Stretch the panel to 100% of the editor container width.
+	 *				this.element.setStyle( 'width', container.getSize( 'width' ) + 'px' );
+	 *			};
+	 *
+	 *			function CustomAutocomplete( editor, textTestCallback, dataCallback ) {
+	 *				// Call the parent class constructor.
+	 *				Autocomplete.call( this, editor, textTestCallback, dataCallback );
+	 *			}
+	 *			// Inherit the autocomplete methods.
+	 *			CustomAutocomplete.prototype = CKEDITOR.tools.prototypedCopy( Autocomplete.prototype );
+	 *
+	 *			CustomAutocomplete.prototype.getView = function() {
+	 *				return new CustomView( this.editor );
+	 *			}
+	 *
+	 *			// Expose the custom autocomplete so it can be used later.
+	 *			CKEDITOR.plugins.customAutocomplete = CustomAutocomplete;
+	 *		}
+	 *	} );
+	 * ```
+	 *
+	 * @class CKEDITOR.plugins.autocomplete
+	 * @since 4.10.0
+	 * @constructor Creates the new instance of autocomplete and attaches it to the editor.
+	 * @param {CKEDITOR.editor} editor The editor to watch.
+	 * @param {Function} textTestCallback Callback executed to check if a text next to the selection should open
```

Comment:
I think there is missing information that `textTestCallback` is called with `range` argument, what is showed in example in general description of the plugin.

Reply:
There is also information: `See the {@link CKEDITOR.plugins.textWatcher}'s `callback` argument` because this callback will be passed directly into `textwatcher`. `textwatcher` already contains information about `range` parameter. I'm not sure if I should duplicate it. WDYT?